src/Symfony/Component/Serializer/Debug/Normalizer/AbstractTraceableNormalizer.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Debug\Normalizer;
+
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\SerializerAwareInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+abstract class AbstractTraceableNormalizer implements SerializerAwareInterface, DenormalizerAwareInterface, NormalizerAwareInterface, CacheableSupportsMethodInterface
+{
+    /**
+     * @var DenormalizerInterface|NormalizerInterface
+     */
+    protected $delegate;
+    private $normalizations = [];
+    private $denormalizations = [];
+
+    /**
+     * AbstractTraceableNormalizer constructor.
+     *
+     * @param DenormalizerInterface|NormalizerInterface $delegate
+     */
+    public function __construct($delegate)
+    {
+        $this->delegate = $delegate;
+    }
+
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        $denormalization = new Denormalization($this->delegate, $data, $type, $format, $context);
+        $denormalization->result = $this->delegate->denormalize($data, $type, $format, $context);
+        $this->denormalizations[] = $denormalization;
```

Comment:
Is it needed to initialize the object before calling the denormalizer?
What's about
```
$result = $this->delegate->denormalize($data, $type, $format, $context);
$this->denormalizations[] = new Denormalization($this->delegate, $data, $type, $format, $context, $result);

return $result;
```

Reply:
Thank you for the suggestion.